### DISCUSSION TITLE
Merking av lenker

### DISCUSSION THREAD

**Discussion by mrosvik** [digdir/designsystemet#2294] [Patterns for public services]
- Når en lenke tar brukeren til en ekstern side, skal vi da vise et ikon eller en tekst i lenken som indikerer at brukeren blir tatt til en ekstern side? 
- Når skal lenken åpnes i samme fane og når skal den åpnes i en ny fane/vindu? 
   - Hvordan informerer vi brukeren om at lenken åpnes i en ny fane?
- Hvordan merker vi filer

**Comment by Barsnes** [comment]
Eg synst ei ekstern lenke skal markerast med ikon av ein boks med ei pil som går ut:
![image](https://github.com/user-attachments/assets/416fcad9-1f1c-42fe-ba54-66cd75e89787)
Dette viser at lenka går ut av "boksen"(sida) du er på. Ikonet burde komme bak lenka:
![image](https://github.com/user-attachments/assets/e958345c-8493-4eba-b481-d37034718ada)

Samtidig synst eg at alle eksterne lenker skal bli åpna i ny fane, så navigeringen brukaren allerede har gjort ikkje blir avbrutt.
Når du går på ei ekstern side er det liten sjanse for at den linker tilbake til sida du var på, og du kan miste kor du var på den original sida.

Eit negativt punkt med denne løysingen er når folk ikkje forstår ikonet. Eg trur ikkje noken blir redde for å trykke på lenka.
Det er allerede eit etablert mønster på internett, men ikkje alle har tatt det i bruk. Dersom me lager eit mønster på det, og det blir følgt, så vil dette bli naturlig for alle brukarar.

**Comment by mrosvik** [comment]
Alternative tekstlige varianter der du forstår at du blir tatt til en ny side: 

- "[Les mer om merverdiavgift på skatteetaten.no](url)"
- "[Gå til skatteetaten.no](url)"

**Reply by solhell** [reply]
På generell basis er jeg ikke fan av lenketekster som begynner med "Les mer", "Klikk her", "Gå til", etc. Jeg mener det er bedre å gå rett på sak i stil med [Merverdiavgift (skatteetaten.no)](https://www.skatteetaten.no/nn/bedrift-og-organisasjon/avgifter/mva) når det er frittstående lenker, eller når i løpende tekst at "les mer" og tilsvarende flyttes ut av lenken slik at det blir les mer om [merverdiavgift (skatteetaten.no)](https://www.skatteetaten.no/nn/bedrift-og-organisasjon/avgifter/mva).

**Comment by Febakke** [comment]
Jeg liker best at lenker som går til andre sider beskrives i selve lenketeksten.
1. Ikonet er tvetydig, brukes både for eksterne linker og åpne i egen fane. 
<img width="771" alt="Skjermbilde 2024-08-21 kl  10 13 16" src="https://github.com/user-attachments/assets/6c911d89-23da-4550-b830-9319ed898ff2">
<img width="843" alt="Skjermbilde 2024-08-21 kl  10 13 08" src="https://github.com/user-attachments/assets/69902624-2887-4466-bcdf-1e3e842f0541">

2. Uansett om alle er klar over det, eller kommer til å bli klar over hva ikonet betyr vil det aldri slå å skrive det i klar tekst. Jeg syns at det skal komme frem fra lenketeksten om den går til et nytt domene. 

Til @Barsnes sitt poeng så skriver tilsynet at man kan bruke ikoner, men går ikke i detalj på hvilken. https://www.uutilsynet.no/veiledning/lenker/214

NAV Aksel anbefaler ikke bruk av ikoner for verken åpne i ny fane eller eksterne sider. 
https://aksel.nav.no/god-praksis/artikler/lenker

Når det gjelder åpne i ny fane/vindu så tror jeg det er viktig når man er i en flyt du ikke vil avbryte. For eksempel fyller ut et skjema. Ellers ville jeg ikke endret forventet oppførsel. De fleste lenker åpner seg i samme vindu som du er i.

**Comment by eirikbacker** [comment]
Litt enig med deg @Febakke - vi hadde akkurat denne diskusjonen på Mattilsynet, for når vi lenker til noen andre, så kan ikke vi stå ansvarlig for kvaliteten på innholdet - det vil vi jo indikere men er ikke nødvendig ny fane.. andre ganger ønsker vi å åpne i ny fane - også til våre egne URLer - for å unngå at brukeren mister hvor de er i prosessen. For å gjøre det heeeeelt trygt for brukeren har vi endt opp med å skrive `Lenketekst (åpner i ny fane)`.

Vi ser også at både eldre brukere på desktop, og de fleste på mobil, blir irritert eller forvirrent av at vi velger å åpne i nye faner, fordi det "brekker" forventet oppførsel til tilbakeknappen.

Dersom vi lander på å tilby et ikon, må vi også alltid ha en `aria-label="Åpnes i ny fane"` for å tilfredsstille UU-krav, så det er noe å ha i bakhodet i forhold til API-utforming også :)

**Reply by Barsnes** [reply]
Einig med at det kan "brekke" navigeringen til brukaren, og eg mistenker at det er derfor Safari har lagt til at tilbakeknappen i ei fane som blei åpna av ein anna tab tar deg til den forrige tabben og lukka fana som blei åpna.
Veit ikkje om andre nettlesarar gjer det samme

**Reply by eirikbacker** [reply]
oi, det viste jeg ikke de gjorde! ikke andre som gjør det meg bekjent :)

**Reply by schjetne** [reply]
Jeg er veldig glad for den funksjonaliteten fordi jeg er en sånn gamling som går til tilbakeknappen når jeg vil *tilbake* der jeg kom fra 😄

**Comment by schjetne** [comment]
Følger denne diskusjonen med interesse, og har også tatt det opp med teamet mitt som bygger Punkt, Oslo Origos designsystem. Jeg er tilbøyelig til å være enig med de som foreslår tekst, og jeg synes tvetydigheten til det "ekstern lenke"-ikonet er frustrerende. Betyr ikonet at det åpnes i ny fane, eller at det er en ekstern lenke? I Punkt bruker vi det til eksterne lenker, men det er vel ingen som er helt fornøyd med det, og litt avhengig av hva konsensus blir i denne diskusjonen er det mulig vi skal vurdere å endre praksis.

**Comment by TomasEng** [comment]
Jeg er av den oppfatning at eksterne lenker alltid burde åpnes i ny fane, mens interne alltid åpnes i den samme. Selv synes jeg i hvert fall at det er frustrerende når interne lenker åpner nye faner, siden jeg da ender opp med mange faner og det blir vanskelig å finne tilbake dit jeg kom fra. Hvis jeg likevel _ønsker_ å åpne noe i en ny fane, vet jeg at jeg for eksempel kan klikke med musehjulet.

Med en slik regel som utgangspunkt opphører tvetydigheten. Man vet at markeringen betyr _både_ at lenken åpnes i en ny fane _og_ at den er ekstern. Når det gjelder markeringen, er ikonet allerede godt etablert, så jeg ser ikke noe videre problem med å bruke det. Men jeg liker også at det står "(ekstern lenke – åpnes i ny fane)" i tittelen på lenken. Jeg er tilhenger av å gjøre begge deler.

**Comment by mrosvik** [comment]
Forslag til foreløpige retningslinjer på `link`-komponenten ang. åpne i ny fane: 

> Både interne og eksterne lenker bør åpnes i samme fane/vindu. Da sikrer vi en forutsigbar brukeropplevelse fordi brukerens egne lenkeinnstillinger i nettleseren videreføres. Flere bruker Tilbake-knappen i nettleseren hvis lenken ikke førte dit de trodde, men det fungerer ikke når lenken ble åpnet i en ny fane.
> 
> Unntaket er hvis brukeren risikerer å miste innhold ved å klikke på lenken, for eksempel ved utfylling av et skjema. Det skal da informeres om at lenken åpnes i en ny fane. 
> 
> [Les mer om lenker på uutilsynet.no](https://www.uutilsynet.no/veiledning/lenker/214)

**Reply by TomasEng** [reply]
Dette med å miste innhold er et godt poeng når det gjelder å åpne interne lenker nye faner. Samtidig vil jeg påstå at dersom man kommer i en situasjon hvor man faktisk risikerer å miste innhold ved å klikke på en lenke, så kan det være et symptom på at noe er galt med måten siden er utformet på.

**Comment by Camulos** [comment]
Vi (Brønnøysundregistrene) har følgende krav som er relevante for merking av lenker:

- Hvis vi lenker til et annet nettsted eller til en fil (ikke-web dokument) er dette vist slik at det er tydelig for brukeren.
- Meningen med en lenke er tydelig: Alle lenker har tekst som sier noe om hva lenken fører til (ikke bare "trykk her" eller "mer").

Krav 2 der er ment å dekke eller få oss til å huske på WCAG suksesskriteriene:

- 2.4.4 Formål med lenke (i kontekst) (A)
- 2.4.9 Link Purpose (Link Only) (AAA)

I praksis så betyr det at lenker som peker til et annet domene, åpnes i ny fane eller er til en fil skal være mulig å forstå før en klikker:

- "denne lenken fører til et tema på denne andre siden (ekstern lenke) _ikon_"
- "denne lenken fører til mer informasjon om det du nå holder på med (åpnes i ny fane) _ikon_"
- "denne lenken fører til mer informasjon på denne andre siden om det du nå holder på med (ekstern lenke, åpnes i ny fane) _ikon_"

Jeg krever ikke at de som bygger tjenester skal alltid bruke både tekst i parentes, ikon og forklare det i lenketeksten eller i setningen/paragrafen hvor lenken står. Men det skal gjøres likt i hele tjenesten og en skal bruke tekst i parentes eller ikon uansett om det er forklart i teksten eller ikke. Det er nok noe mer naturlig med kun ikon i tillegg om det allerede står i teksten at det er en ekstern side eller at lenken ikke avbryter det brukeren holder på med (feks fylle ut et skjema, som er eneste konteksten vi ikke åpner i samme fane).

Per i dag er det gjort en god del arbeid på å i hvert fall markere så mange eksterne lenker som mulig med ikon. Åpne i ny fane er kun tillatt i skjema hos oss og det skal det også ha vært gjort en god arbeid med å fikse.

Eksempel på lenker til annet domene har vi ofte i "Nyttige lenker" boks på hjelpeartikler, og når vi feks peker til et skjema på Altinn.
![eksempelGrasrotandelen-artikkelBrreg](https://github.com/user-attachments/assets/decfe1da-8423-47d6-8edc-9c95aea8621d)
![eksempelSORM-eksternlenke](https://github.com/user-attachments/assets/809b8fd7-0010-4642-b6de-7a39dcd98373)

**Comment by mrosvik** [comment]
Vi starter opp dette mønsteret Januar 2025. Digdir inviterer og fasilitererer.

**Comment by Mokkatroffen** [comment]
@Camulos:
Eksempelet deres fra "Nyttige lenker" bryter med lovkravene (1.1.1) om man ikke tilfører ytterligere informasjon. Jeg synes også at det er litt snodig hvordan ikonet ikke er interaktivt / en del av lenke definisjonen. 

Hvis dere ikke setter et aboslutt krav om å skrive (åpnes i ny fane) (last ned pdf) etc på enden av lenker må dere stille krav om at slike ikoner har et tekstlig alterantiv. I praksis vil en seende i dag få vite at disse lenkene åpner ny fane, skjermleserbrukere vil ikke få denne informasjonen:
![image](https://github.com/user-attachments/assets/7fa4986a-b068-4e7e-9549-328d138151ca)

Merk også at det her er flere skjermleser aspekter som burde vurderes, som hvorvidt informasjonen ikonet kommuniserer er tilgjengelig i elementlisten til ulike skjermlesere, hvorvidt dere skal bruke ikon/bilde og hvorvidt den beskrivende teksten skal være en del av denne definisjonen eller defineres på en annen måte (sr-only).

**Comment by solhell** [comment]
Mitt forslag er som følger:
Lenketekst som kort forteller hva som er innholdet på siden det lenkes til + (domenenavn) + ikon [alt-tekst: åpens i ny fane]

For eksempel: [Komponenter i designsystemet (designsystemet.no)](https://www.designsystemet.no/komponenter) (og så ikon til slutt med nevnte alt-tekst, dette må ligge inn i (a)-elementet slik at det blir lest opp av skjermleser)

Jeg mener det gir merverdi å fortelle domenenavnet i stedet for bare den generiske "ekstern lenke".

**Comment by sigveres** [comment]
Kommer inn litt sent i diskusjonen her, men deler en teknikk som jeg har brukt for lenker i løpende tekst. Bruker denne teknikken i Link-komponent for Patentstyrets designsystem.

**Ikon**
Vi bruker ikon for både interne og eksterne lenker for å visuelt vise hensikten med lenke for alle brukere, uavhengig funksjonsnedsettelse.
Ikonet gir lenker en forsterket visuell effekt som kan være nyttig for alle. Bruken av ikon støtter også [WCAG 1.4.1 -  bruk av farge](https://www.uutilsynet.no/veiledning/141-bruk-av-farge/1239) fordi det gir et ekstra hint enn kun understrek alene.

**Tilgjengelighet**
For at "åpne i ny fane" skal leses opp korrekt av skjermleser bruker jeg `aria-describedby` som peker til en tekst med `sr-only`-klasse. På den måten blir informasjonen behandlet som _tilleggsinformasjon_, og ikke en del av lenkens navn - slik det ville bli gjort med `aria-label`. 
Denne teknikken funker også dersom ikon blir lagt inn med css, eller om ikoner by default er skjult. 

**Eksempel**
``` HTML
<a href="#"  target="_blank" rel="noopener noreferrer" aria-describedby="external-desc-1">
  Nyttig lenke
  <span aria-hidden="true" class="icon">open_in_new</span>
</a>

<span id="external-desc-1" class="sr-only">
  Åpner i ny fane
</span>
```

**Potensielt problem med WCAG 2.5.3 Ledetekst i navn**
Usikker om dette er et problem enda, men en annen grunn til at jeg har unngått å overskrive ikke-visuelle aria-label-verdier da det kan gi konflikter med [WCAG 2.5.3 Ledetekst i navn](https://www.uutilsynet.no/wcag-standarden/253-ledetekst-i-navn-niva/150) opp mot talegjenkjenning.

`Aria-label` erstatter den synlige teksten som _tilgjengelig navn_ for tilgjengelighetsverktøy. Dette skaper inkonsekvens mellom det brukeren
ser og det som talegjenkjenningsprogrammer forventer når de navigerer basert på synlig tekst.
Det vil si at navigering til f.eks lenker ikke vil funke, fordi talegjenkjenningsprogrammer forventer at du skal si "[lenketekst] åpner i ny fane", ikke bare "[lenketekst]"

**Eksempel med aria-label**
Eksempelet viser at lenkenavnet egentlig er "Nyttig lenke - åpner i ny fane" selv om den visuelle lenken er "Nyttig lenke"
<img width="511" height="491" alt="" src="https://github.com/user-attachments/assets/968ae7dc-1bb1-42f0-96d9-d052c9a18a7f" />


**Eksempel med aria-description**
Eksempelet viser at lenkenavnet kun er "Nyttig lenke" som er lik den visuelle teksten
<img width="531" height="486" alt="" src="https://github.com/user-attachments/assets/7b4435f2-ced6-4c8b-b6c0-428cdaa187f1" />

**Reply by mrosvik** [reply]
Takk for godt innspill. Dette har vi vurdert, og vi har landet på å ikke anbefale bruk av ikon.

Grunnen er at dette ikonet blir tolket svært ulikt av brukere. Noen tror det betyr at lenken åpnes i ny fane, andre at det betyr ekstern side, og noen oppfatter det bare som en dekorasjon. Dermed gir ikke ikonet i seg selv tydelig og forutsigbar informasjon.

Derfor anbefaler vi heller at:
- Informasjon om at en lenke er ekstern, eller åpnes i ny fane, formidles i selve lenketeksten der det er relevant.
- Ikon ikke skal være eneste bærer av denne informasjonen, av hensyn til både forståelse og tilgjengelighet.

Vi har samlet vurderingene våre her: https://designsystemet.no/no/patterns/external-links

Vi tar gjerne imot flere perspektiver, men dette er bakgrunnen for anbefalingen slik den er formulert nå.

**Comment by mrosvik** [comment]
Kommentar på UX Norge 19.06.2026:

> Det finnes jo et ikon som ofte blir brukt til å indikere at man kan åpne i ny fane, men Designsystemet har [en god del av en artikkel ](https://designsystemet.no/no/patterns/external-links#unng%C3%A5-ekstern-lenke-ikon)om hvorfor «ekstern lenke»-/«åpne i ny fane»-ikonet er tvetydig. Men artikkelen er kanskje ikke helt tydelig på om den anbefaler å ikke bruke det i det hele tatt, eller om den bare anbefaler at man ikke skal bruke ikonet til eksterne lenker ([@Marianne Røsvik](https://designsystemet.slack.com/team/U05S5RY6Q80), kanskje det er en mulig upgrade av artikkelen). Hvis alle i hele det offentlige Norge tenker at dette ikonet skal brukes, men kun til å åpne i ny fane, så er jo det superpraktisk retningslinje for oss.
